docs/fieldnorm/requirements.md

```diff
@@ -0,0 +1,20 @@
+# Requirements in software development
+The complexity and variety of developments in the field of bioinformatics, comes with a need for clear prerequisites before the start of a project. These requirements of software should describe the prerequisites that the software should meet and they should be clear before starting on a project. The requirements may relate to several aspects, including:
+- the performance of the desired software
+- the physical characteristics
+- the computing environment
+- a need for compatibility
+- input- and output-formats
+- interfaces to other systems
+- communication from the software
+- security requirements
+- user interfaces to the software
+- data definitions
+- installation and acceptance requirements
+- requirements related to the methods of operation and maintenance
+- network-related requirements
+- user maintenance requirements
+- regulatory requirements
+
+All software development projects should have a standardized and documented approach for gathering requirements. Requirements can come from end-users, bio-informaticians and other stakeholders. It is recommended to express the requirements in a form that is SMART (specific, measurable, attainable, realistic and time-bound). It is essential to document these requirements clearly and track them throughout the development process (e.g. in JIRA, Confluence or GitHub/GitLab). Once these requirements have been implemented as features in the software, they should be validated before the release is used in production. Any changes or updates to these requirements should be carefully reviewed and documented.
```

Comment:
> All software development projects should have a standardized and documented approach for gathering requirements.

So if one of the lab analysts come to my desk and asks me to add a feature. I cannot simply add this as a gitlab issue and be done with it? I need to write that in documentation?

Reply:
I feel like this shouldn't be "all software projects" as this would imply a different way of submitting/gathering requirements for each different project which should than also be documented every time. Maybe it is better to say "Bioinformatic teams should have a..." this would make it more standardized and not having to write a document for every other project.

Reply:
> > All software development projects should have a standardized and documented approach for gathering requirements.
> 
> So if one of the lab analysts come to my desk and asks me to add a feature. I cannot simply add this as a gitlab issue and be done with it? I need to write that in documentation?

'to document these requirements clearly and track them throughout the development process (e.g. in JIRA, Confluence or GitHub/GitLab).' Why would this stop you for documenting this in Issue form? It all depends on your setup how to handle these cases.

Reply:
If you add requirements to gitlab, you have no clue whether it's already stated somewhere.
Requirements don't change much, make sure they are not too detailed